pennylane/tape/tape.py

```diff
@@ -872,6 +872,145 @@ def set_parameters(self, params, trainable_only=True):
             op = self._par_info[idx]["op"]
             op.data[self._par_info[idx]["p_idx"]] = p
 
+    @staticmethod
+    def _single_measurement_shape(measurement_process, device):
+        """
+
+        Note: these shapes and dtypes are specific to tapes and may differ from
+        QNode outputs.
+        """
+
+        shape = tuple()
+
+        output_domain = "real"
+        ret_type = measurement_process.return_type
+        if ret_type in (qml.operation.Expectation, qml.operation.Variance):
+
+            shape = (1,)
```

Comment:
Hmm, this definitely feels like something that should be a property of the return type/measurement 🤔

Reply:
Thanks! I've moved the fixed ones to the `MeasurementProcess` class and created a `shape` kwarg in general. For some, we need extra info to determine the shape (e.g., num wires or the device).